draft-ietf-jsonpath-base.md

```diff
@@ -209,7 +209,9 @@ Argument:
 : Short name for the value a JSONPath expression is applied to.
 
 Node:
-: The pair of a value along with its location within the argument.
+: The pair of a value along with its location within the argument
+  except for member names which do not have nodes. (Members are not
```

Comment:
Maybe we can separate this out into 
-- member names do not have nodes
-- members aren't values in the first place, so they also do not have nodes
Maybe we can also cross-ref "Child".

Reply:
and ... what exactly is `its location`? 

If it's the member name or element index, we should say so. If it's not, we need to define it.

Reply:
The word location is used in its English sense here.
A location can be represented using a normalized path (which is a representation of a sequence of edges in the tree that a JSON value is).

Reply:
Hmm ... maybe this definition qualifies to be inserted in the Terminology section.

Reply:
I considered separate sentences for member names and members, but that could contradict the first sentence of the definition of node. Member names, as you point out, are values and they *might* have locations (since we don't actually define what a location is!). Hence my use of "except for".

We could _note_ that a node is associated either with the argument or one of its descendants. But we can't actually define a node that way because the definition would be circular.

Reply:
This is quite complicated. What about:

```
... except for member names which are no JSON values.
```

Reply:
@goessner @cabo pointed out that members names are strings which _are_ JSON values.

Reply:
Hmm ... so our members as `name/value` pairs are de facto `value/value` pairs? 

I would prefer to define a member name as a `string` in the sense of JSON and being quiet about the confusing logical conclusion, that it must be a JSON value then ... but of course without being a structured value (potentially having nodes) or any other primitive type.

Reply:
Ok, I defined "Location" as a term, split out the statements about member names and members not being nodes, and cross referenced descendants from the definition of node.